plugins/rbac/src/components/RolesList/RolesList.tsx

```diff
@@ -42,6 +44,16 @@ export const RolesList = () => {
     setRoles(searchResults.length);
   };
 
+  React.useEffect(() => {
+    let mounted = false;
+    if (!mounted && !loading) {
+      setIsMounted(true);
+    }
+    return () => {
+      mounted = true;
+    };
+  }, [loading]);
+
```

Comment:
The condition `if (!mounted && !loading) ` will always be true when `loading` is false as `mounted` is being initialized to `false`. Since `mounted` here is a local variable within each render, changes made by the cleanup function will not persist across renders. How about we use `useRef` instead?

```
import React, { useEffect, useRef, useState } from 'react';

export const RolesList = () => {
  // ...
  const [isMounted, setIsMounted] = useState(false);
  const mountedRef = useRef(false);

  useEffect(() => {
    mountedRef.current = true;
    if (!loading) {
      setIsMounted(true);
    }
    return () => {
      mountedRef.current = false;
    };
  }, [loading]);

  // ...
};
```